tap_zendesk/streams.py

```diff
@@ -96,6 +97,20 @@ def load_metadata(self):
     def is_selected(self):
         return self.stream is not None
 
+def raise_or_log_zenpy_apiexception(schema, e):
+    # There are multiple tiers of Zendesk accounts. Some of them have
+    # access to `custom_fields` and some do not. This is the specific
+    # error that appears to be return from the API call in the event that
+    # it doesn't have access.
+    if not isinstance(e, zenpy.lib.exception.APIException):
+        raise ValueError("Called with a bad exception type") from e
+    if json.loads(e.args[0])['error']['message'] == "You do not have access to this page. Please contact the account owner of this help desk for further help.":
+        LOGGER.warning("The account credentials supplied do not have access to custom fields.")
```

Comment:
This message comes through whenever the user's token doesn't have access to whichever feature is requested. Do you think it would be okay to pass the "type" of feature into this function to be used in the message?

Reply:
Yeah that sounds great. :)

I'll do it in the morning.